protocol/bc/bc.proto

```diff
@@ -20,3 +25,148 @@ message AssetID {
   fixed64 v2 = 3;
   fixed64 v3 = 4;
 }
+
+message AssetAmount {
+  AssetID asset_id = 1;
+  uint64  amount   = 2;
+}
+
+message AssetDefinition {
+  Hash    initial_block_id = 1;
+  Program issuance_program = 2;
+  Hash    data             = 3;
+}
+
+message ValueSource {
+  Hash        ref      = 1;
+  AssetAmount value    = 2;
+  uint64      position = 3;
+}
+
+message ValueDestination {
+  Hash        ref      = 1;
+  AssetAmount value    = 2;
+  uint64      position = 3;
+}
+
+message BlockHeaderEntry {
+  message Body {
+    uint64 version                = 1;
+    uint64 height                 = 2;
+    Hash   previous_block_id      = 3;
+    uint64 timestamp_ms           = 4;
+    Hash   transactions_root      = 5;
+    Hash   assets_root            = 6;
+    bytes  next_consensus_program = 7;
+    Hash   ext_hash               = 8;
+  }
+  message Witness {
+    repeated bytes arguments = 1;
+  }
+  Body    body    = 1;
+  Witness witness = 2;
+}
+
+message TxHeader {
+  message Body {
+    uint64        version     = 1;
+    repeated Hash result_ids  = 2;
+    Hash          data        = 3;
+    uint64        min_time_ms = 4;
+    uint64        max_time_ms = 5;
+    Hash          ext_hash    = 6;
+  }
+  Body body = 1;
+}
+
+message Mux {
+  message Body {
+    repeated ValueSource sources  = 1; // issuances, spends, and muxes
+    Program              program  = 2;
+    Hash                 ext_hash = 3;
```

Comment:
Mostly by hand, partly by Emacs. It would be nice (and wouldn't be hard) to write a `protofmt` tool, now that you mention it.